plugins/kubernetes/app/models/kubernetes/role.rb

```diff
@@ -128,9 +128,12 @@ def seed_service_name(config_file)
     # TODO: support dynamic folders
     def defaults
       return unless resource = role_config_file('HEAD', deploy_group: nil, ignore_errors: true)&.primary
-      spec = resource.fetch(:spec)
+      spec = resource.fetch(:spec, {})
       if resource[:kind] == "Pod"
         replicas = 0 # these are one-off tasks most of the time, so we should not count them in totals
+      elsif resource[:kind] == "PodTemplate"
+        replicas = 1
+        spec = resource.dig(:template, :spec) || {}
       else
         replicas = spec[:replicas] || 1
         spec = spec.dig(:template, :spec) || {}
```

Comment:
```suggestion
      if resource[:kind] == "Pod"
        replicas = 0 # these are one-off tasks most of the time, so we should not count them in totals
        spec = resource.fetch(:spec)
      elsif resource[:kind] == "PodTemplate"
        replicas = 1
        spec = resource.dig(:template, :spec) || {}
      else
        replicas = spec[:replicas] || 1
        spec = resource.dig(:spec, :template, :spec) || {}
```

Reply:
also this duplicate the logic used to find out if it's `primary`, so best reuse that here ?

Reply:
Logically yes. Semantically? Not quite sure. This isn't my area of expertise and my knowledge is very limited. I'm happy for you suggest a variant that utilises primary instead, as I don't see how it can be used.

Reply:
I'm sorry but what you have suggested does not work. In the case of deployments and other resources with embedded templates, we need to retrieve the the replicas from the first instance of spec, before diving deeper into the template spec. The implementation outlined in this PR accounts for this.

I'll keep digging into the primary thing. Sorry, kind of new to this space.

Reply:
ah yes, they are different specs ... might need some better variable nameing there too
don't worry about the primary, I can clean that up after this PR is done

Reply:
Updated your logic to work as intended. Thanks for the suggestion ❤️

Reply:
I'll let you handle the primary refactor later on if needed.